### PR TITLE
Support remote credentials for push/pull

### LINKED ISSUE #7: Support remote credentials for pull/push

### PR DESCRIPTION
**Pull request by atugushev**
Fixes #7.

### CODE REVIEW

File: wok/core/base.py [hunk 1]
```diff
@@ -128,3 +129,35 @@ def finish(repo: pygit2.Repository, branch_name: str, message: str) -> None:
...
+            username = username_from_url or input("Username: ")
+            password = getpass.getpass()
+            return pygit2.UserPass(username=username, password=password)
+        elif allowed_types & pygit2.credentials.GIT_CREDTYPE_USERNAME:
+            return pygit2.Username(username_from_url)
+        else:
+            return None
+
+
+def clone(url: str, path: str, **kwargs: typing.Any) -> pygit2.Repository:
```
Comment (lig) [inline wok/core/base.py]: It looks like it's handier to use `pathlib.Path` for `path` here.

File: wok/core/base.py [hunk 2]
```diff
@@ -128,3 +129,35 @@ def finish(repo: pygit2.Repository, branch_name: str, message: str) -> None:
...
+    ]:
+        """
+        If the remote server requires authentication, this function will be called and
+        its return value used for authentication.
+        """
+        if allowed_types & pygit2.credentials.GIT_CREDTYPE_SSH_KEY:
+            return pygit2.KeypairFromAgent(username_from_url)
+        elif allowed_types & pygit2.credentials.GIT_CREDTYPE_USERPASS_PLAINTEXT:
+            username = username_from_url or input("Username: ")
+            password = getpass.getpass()
```
Comment (lig) [inline wok/core/base.py]: I'm tempted to use `click`'s password prompt here. However, I see no real reason for that. So, just a thought.

**Review by lig — CHANGES_REQUESTED** [review]

File: wok/core/base.py (hunk 1 above)
Comment (atugushev) [reply wok/core/base.py]: Right! Addressed in b07298edeccd4e7ceacd3de2da1388ac28f54f93.

File: wok/core/base.py (hunk 2 above)
Comment (atugushev) [reply wok/core/base.py]: Agree with you. Addressed in 741facc.

**Pull request merged**